[webkitbugspy] Get and set issue component data

JonWBedard opened the pull request:
#### 61ca77011cb9ddc963017edb372dc4d97d7e4cd1
<pre>
[webkitbugspy] Get and set issue component data
<a href="https://bugs.webkit.org/show_bug.cgi?id=238514">https://bugs.webkit.org/show_bug.cgi?id=238514</a>
&lt;rdar://problem/90994542 &gt;

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker.populate): Populate project, component and version.
(Tracker.set): Set project, component and version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.request): github.Tracker has no Exception member. Use RuntimeError instead.
(Tracker.populate): Populate labels and then extract component and version.
(Tracker.set): Set labels associated with components and versions.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py:
(Issue.__init__): Set _labels, _project, _component and _version values.
(Issue.labels): Populate and return labels.
(Issue.set_labels): Set labels associated with issue.
(Issue.project): Populate and return project.
(Issue.component): Populate and return component.
(Issue.version): Populate and return version.
(Issue.set_component): Set the project, component and version of an issue.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla._issue): Mock setting and returning project, component and version.
(Bugzilla._create): Populate project, component and version on creation.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py: Set values for
project, component and version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py:
(GitHub._labels_for_issue): Given an issue, return a list of labels. Take into
consideration component and version.
(GitHub._issue): Mock label editing and returning.
(GitHub._create): Ditto.
(GitHub.request): Support request to update labels.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py:
(RadarModel.__init__): Set component.
(RadarModel.commit_changes): Commit component changes.
(RadarClient.find_components): Implement version filter.
(RadarClient.create_radar): Create radar with component details.
(Radar.Component.get): Implement dict-like API.
(Radar.Component.__getitem__): Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker.populate): Get project, component and version.
(Tracker.set): Set component information.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
(TestBugzilla.test_create):
(TestBugzilla.test_create_prompt):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py:
(TestGitHub.test_projects):
(TestGitHub.test_create_projects):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/radar_unittest.py:
(TestRadar.test_create):
(TestRadar.test_create_prompt):


Canonical link: <a href="https://commits.webkit.org/249039@main">https://commits.webkit.org/249039@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292132">https://svn.webkit.org/repository/webkit/trunk@292132</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

dewei-zhu commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
```diff
@@ -138,7 +138,7 @@ def request(self, path=None, params=None, headers=None, authenticated=None, pagi
             auth = None
         if authenticated and not auth:
-            raise self.Exception('Request requires authentication, none provided')
+            raise RuntimeError('Request requires authentication, none provided')
```
Why do we make this change? I don't find it in commit message.

JonWBedard replied:
Will add it. We don't actually have a `self.Exception` in this class. The code was moved from a class that did have this.

dewei-zhu reviewed (approved):
r=me with comment

JonWBedard committed:
[webkitbugspy] Get and set issue component data
https://bugs.webkit.org/show_bug.cgi?id=238514
<rdar://problem/90994542>

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker.populate): Populate project, component and version.
(Tracker.set): Set project, component and version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.request): github.Tracker has no Exception member. Use RuntimeError instead.
(Tracker.populate): Populate labels and then extract component and version.
(Tracker.set): Set labels associated with components and versions.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py:
(Issue.__init__): Set _labels, _project, _component and _version values.
(Issue.labels): Populate and return labels.
(Issue.set_labels): Set labels associated with issue.
(Issue.project): Populate and return project.
(Issue.component): Populate and return component.
(Issue.version): Populate and return version.
(Issue.set_component): Set the project, component and version of an issue.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla._issue): Mock setting and returning project, component and version.
(Bugzilla._create): Populate project, component and version on creation.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py: Set values for
project, component and version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py:
(GitHub._labels_for_issue): Given an issue, return a list of labels. Take into
consideration component and version.
(GitHub._issue): Mock label editing and returning.
(GitHub._create): Ditto.
(GitHub.request): Support request to update labels.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py:
(RadarModel.__init__): Set component.
(RadarModel.commit_changes): Commit component changes.
(RadarClient.find_components): Implement version filter.
(RadarClient.create_radar): Create radar with component details.
(Radar.Component.get): Implement dict-like API.
(Radar.Component.__getitem__): Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker.populate): Get project, component and version.
(Tracker.set): Set component information.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
(TestBugzilla.test_create):
(TestBugzilla.test_create_prompt):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py:
(TestGitHub.test_projects):
(TestGitHub.test_create_projects):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/radar_unittest.py:
(TestRadar.test_create):
(TestRadar.test_create_prompt):
